### PR TITLE
Parse toml config table into struct

### PR DESCRIPTION
**Pull request by mlange-42**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -288,50 +286,49 @@ impl Preprocessor for Bibiography {
...
 
-    fn run(&self, ctx: &PreprocessorContext, mut book: Book) -> Result<Book, Error> {
+    fn run(&self, ctx: &PreprocessorContext, mut book: Book) -> Result<Book, anyhow::Error> {
         info!("Processor Name: {}", self.name());
 
-        if let Some(cfg) = ctx.config.get_preprocessor(self.name()) {
-            let bib_content = Bibiography::retrieve_bibliography_content(ctx, cfg);
-
-            if bib_content.is_err() {
+        let config = match Config::try_from(ctx.config.get_preprocessor(self.name())) {
```
Comment (mlange-42) [inline src/lib.rs:292]: Not sure here. Should we generally only warn and skip on errors?

Comment (francisco-perez-sorrosal) [reply src/lib.rs:292]: Let's leave the warning for now and let's point to the README.md of the project. BTW, I forgot to mention in #9, do you want to add some info about the cited_only option in the README.md?

**Review by francisco-perez-sorrosal — APPROVED** [review]
+1

File: README.md [hunk 2]
```diff
@@ -67,4 +66,22 @@ following syntax:
...
+``````
+
+## Configuration
+
+A complete `preprocessor.bib` section with all possible options:
+
+```toml
+[preprocessor.bib]
+bibliography = "my_biblio.bib"
+renderer = ["html"]
```
Comment (mlange-42) [inline README.md:78]: Is `renderer` actually used anywhere? I think renderer support is checked via command line arguments, not via the configuration.

Comment (francisco-perez-sorrosal) [reply README.md:78]: Actually no, I think I had to specify for the preprocessors but it's not required as far as I've seen.

File: README.md [hunk 3]
```diff
@@ -67,4 +66,22 @@ following syntax:
...
+render-bib = "cited"
+```
+
+| Option           | Description                                                             | Default |
+|------------------|-------------------------------------------------------------------------|---------|
+| `bibliography`   | `.bib` file to use                                                      | -       |
+| `zotero_user_id` | Zotero user ID, alternative to bib file                                 | -       |
+| `renderer`       | ???                                                                     | -       |
```
Comment (francisco-perez-sorrosal) [inline README.md:86]: You can remove also this, as per the comment above

File: README.md [hunk 4]
```diff
@@ -38,13 +37,13 @@ The bibliography will appear as a separate section in your book ToC.
...
-section: 
+Alternatively, you can use any publicly available library in BibLaTex format from Zotero.
+In order to do so, just specify the `Zotero UserId` of the public bibliography you want to access in the preprocessor section:
 
 ```toml
 [book]
 #...
+[preprocessor.bib]
 zotero_user_id = __a_Zotero_userID__
 renderer = ["html"]
```
Comment (francisco-perez-sorrosal) [inline README.md:48]: Also this can be removed as per the comment below

**Comment by francisco-perez-sorrosal** [comment]
+1

**Pull request merged**